compiler/src/codegen/transl_anf.re

```diff
@@ -625,27 +626,34 @@ let transl_attributes = attrs => {
 let rec compile_comp = (env, c) => {
   let desc =
     switch (c.comp_desc) {
-    | CSwitch(arg, branches) =>
+    | CSwitch(arg, branches, partial) =>
       let compiled_arg = compile_imm(env, arg);
       let switch_type =
         Option.fold(
           ~none=I32Type,
           ~some=((_, exp)) => asmtype_of_alloctype(exp.anf_allocation_type),
           List.nth_opt(branches, 0),
         );
+      let default =
+        switch (partial) {
+        | Partial => [
+            {
+              instr_desc: MError(Runtime_errors.MatchFailure, [compiled_arg]),
```

Comment:
Like I mentioned in the description, this isn't actually the error that's happening on a match failure. I did have to do this (sensible) change though to allow the exception handler to use `match` and not depend on itself.